gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -234,6 +238,26 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  // DirectPath should only be used on Compute Engine.
+  // Notice Windows is supported for now.
+  static boolean isOnComputeEngine() {
+    String osName = System.getProperty("os.name");
+    if ("Linux".equals(osName)) {
+      String cmd = "cat /sys/class/dmi/id/product_name";
+      try {
+        Process process = Runtime.getRuntime().exec(new String[] {"/bin/sh", "-c", cmd});
```

Comment:
What is going on here? Why is this forking a process, to call a _shell_ to then run the cat command to read a file? That seems unnecessary on two levels. Why not use a `FileInputStream`? Is there something I'm missing?

Reply:
Hey Eric, yes `FileInputStream` can definitely work. Sorry I am not a java expert and I directly follow the doc to use a shall command. Do you want me to make the change?

Reply:
I'm not a gax-java maintaner, so "I have no power here." But it seems like a good idea to fix it, as it reduces the chances of something going wrong and makes it easier to debug. For example, if there's permission problems the error will be more obvious.

Reply:
OK, I will have another PR for this.

Reply:
I think we should fix this. Process/exec on "bin/sh" seems completely unnecessary

Reply:
Please file an issue

Reply:
Done https://github.com/googleapis/gax-java/issues/1323